include/slam_toolbox/laser_utils.hpp

```diff
@@ -75,8 +76,9 @@ class LaserMetadata
 class LaserAssistant
 {
 public:
+  template<class NodeT>
   LaserAssistant(
-    rclcpp::Node::SharedPtr node, tf2_ros::Buffer * tf,
+    std::shared_ptr<NodeT> node, tf2_ros::Buffer * tf,
```

Comment:
The NodeT itself can be a pointer type, that way it can be either a raw, shared, unique, or any other kind of non-weak pointer to a node object.

Reply:
https://github.com/SteveMacenski/slam_toolbox/pull/561/commits/37648d73307b3480a422282e1e4686dee9d4f6b8

Changed `std:shared_ptr<NodeT>` to `NodeT`.